model/SearchEngine/Driver/Elasticsearch/ElasticSearch.php

```diff
@@ -177,6 +178,19 @@ public function index($documents = []): int
             ? $documents
             : new ArrayIterator($documents);
 
+        foreach ($documents as $document) {
+            if (!$document instanceof IndexDocument) {
+                throw new \InvalidArgumentException(
+                    'Expected an instance of IndexDocument, got ' . get_class($document)
+                );
+            }
+            $indexName = $this->indexer->getIndexNameByDocument($document);
+            if (!$this->isExistingIndex($indexName)) {
+                $this->createIndexes();
+                continue;
+            }
```

Comment:
Isn't that all documents are for one index? So we can move this block out of foreach loop?

Reply:
ok so its mix of documents from different indexes gotcha